### PR TITLE
Port fixes and changes from the jitaas branch

### PR DESCRIPTION
**Pull request by ymanton**
This PR contains a few fixes and self-contained changes from the `jitaas` branch.

The biggest patch is https://github.com/eclipse/openj9/commit/45e095255323374ce4255bb7473a9f44dce62c0c, which moves the hints API from the frontend to `TR_J9SharedCache`.

### CODE REVIEW

File: runtime/compiler/codegen/J9AheadOfTimeCompile.cpp [hunk 1]
```diff
@@ -1895,12 +1895,8 @@ J9::AheadOfTimeCompile::dumpRelocationData()
 
          default:
             traceMsg(self()->comp(), "Unknown Relocation type = %d\n", kind);
-            TR_ASSERT(false, "should be unreachable");
-            printf("Unknown Relocation type = %d\n", kind);
-            fflush(stdout);
-            exit(0);
-
-        break;
+            TR_ASSERT_FATAL(false, "should be unreachable");
```
[the remark this answers is not in the document]
Comment (ymanton) [reply runtime/compiler/codegen/J9AheadOfTimeCompile.cpp]: Done.

File: runtime/compiler/env/J9SharedCache.hpp [hunk 2]
```diff
@@ -44,8 +44,11 @@ class TR_J9SharedCache : public TR_SharedCache
 
    bool isHint(TR_ResolvedMethod *, TR_SharedCacheHint, uint16_t *dataField = NULL);
    bool isHint(J9Method *, TR_SharedCacheHint, uint16_t *dataField = NULL);
+   bool isHint(J9ROMMethod *, TR_SharedCacheHint, uint16_t *dataField = NULL);
```
[the remark this answers is not in the document]
Comment (ymanton) [reply runtime/compiler/env/J9SharedCache.hpp]: They don't appear to exist anymore in the `jitaas` branch, probably deleted but the prototypes were forgotten. I've removed them from the patch as well.

**Comment by ymanton** [comment]
@mpirvu, comments addressed.

**Comment by mpirvu** [comment]
@ymanton If there is no more commit to add, could you please remove "WIP" from the title?

**Comment by mpirvu** [comment]
Jenkins test sanity all jdk8,jdk11

**Comment by mpirvu** [comment]
Jenkins test sanity all jdk8,jdk11

**Comment by mpirvu** [comment]
All tests passed, except the test for x86-64 windows on Java11 which was aborted by infra.

**Pull request merged**